mbuild/compound.py

```diff
@@ -774,6 +782,43 @@ def add(self, new_child, label=None, containment=True, replace=False,
                 new_child.periodicity.any()):
             self.periodicity = new_child.periodicity
 
+        # If parent has no box --> inherit child box
+        # If parent has box --> keep unless inherit_box == True
+        # If inherit_box == True, parent box != None, child_box == None,
+        # keep parent box anyway and warn
+        # If inherit_box == False, child.box != None, warn
+        if self.box is None:
+            if new_child.box is not None:
+                self.box = new_child.box
+        else:
+            if inherit_box:
+                if new_child.box is None:
+                    warn(
+                        "The Compound you are adding has no box but "
+                        "inherit_box=True. The box of the original "
+                        "Compound will remain unchanged."
+                    )
+                else:
+                    self.box = new_child.box
+            else:
+                if new_child.box is not None:
+                    warn(
```

Comment:
I guess this warning is sufficient, since comparing maxs and mins directly lead to the question about non-orthogonal box. We should keep a note (may be just a comment line above the code), that comparing the length of the box and bounding box is not a complete fool-proof, since if the compound is not in the center of the box, there would still be case where the length of the bounding box is smaller than the box, but a few particle is still outside. Nonetheless, I think this is the issue to be dealt with for a different PR (updating writers)